.github/workflows/pr.yaml

```diff
@@ -115,18 +115,50 @@ jobs:
           echo "out=${action_files}" >> "$GITHUB_OUTPUT"
 
   # Run tests against all linters for known_good_version and latest version
-  linter_tests:
-    name: Linter Tests ${{ matrix.os }}
-    runs-on: [self-hosted, "${{ matrix.os }}"]
+  linter_tests_linux:
+    name: Linter Tests Linux
+    runs-on: public-amd64-2xlarge
+    needs: detect_changes
+    if:
+      needs.detect_changes.outputs.linters == 'true' || needs.detect_changes.outputs.all-linters ==
```

Comment:
If we end up keeping this, can we merge it back into the matrix and add a new field to the object for the correct runner name?